owncloudData/src/test/java/com/owncloud/android/data/sharing/shares/repository/OCShareRepositoryTest.kt

```diff
@@ -19,497 +19,281 @@
-            "whoever",
-            -1,
-            accountName
-        )
+        val sharesResult = ocShareRepository.getSharesAsLiveData(filePath, OC_ACCOUNT_NAME)
+        assertEquals(sharesLiveDataList, sharesResult)
 
         verify(exactly = 1) {
-            remoteShareDataSource.insert(
-                filePath,
-                ShareType.GROUP,
-                "whoever",
-                -1,
-                accountName = accountName
+            localShareDataSource.getSharesAsLiveData(
+                filePath = filePath,
+                accountName = OC_ACCOUNT_NAME,
+                shareTypes = listOf(ShareType.PUBLIC_LINK, ShareType.USER, ShareType.GROUP, ShareType.FEDERATED)
             )
         }
-
-        verify(exactly = 1) { localShareDataSource.insert(shares[2]) }
     }
 
-    @Test(expected = FileNotFoundException::class)
-    fun insertPrivateShareFileNotFoundException() {
+    @Test
+    fun `getShareAsLiveData returns a LiveData with an OCShare`() {
+        val shareLiveData: LiveData<OCShare> = MutableLiveData(OC_SHARE)
+
         every {
-            remoteShareDataSource.insert(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any()
-            )
-        } throws FileNotFoundException()
+            localShareDataSource.getShareAsLiveData(OC_SHARE.remoteId)
+        } returns shareLiveData
 
-        ocShareRepository.insertPrivateShare(
-            filePath,
-            ShareType.GROUP,
-            "whoever",
-            -1,
-            accountName
-        )
+        val shareResult = ocShareRepository.getShareAsLiveData(OC_SHARE.remoteId)
+        assertEquals(shareLiveData, shareResult)
 
         verify(exactly = 1) {
-            remoteShareDataSource.insert(
-                filePath,
-                ShareType.GROUP,
-                "whoever",
-                -1,
-                accountName = accountName
-            )
-        }
-
-        verify(exactly = 0) {
-            localShareDataSource.insert(shares[2])
+            localShareDataSource.getShareAsLiveData(OC_SHARE.remoteId)
         }
     }
 
     @Test
-    fun updatePrivateShareOk() {
+    fun `refreshSharesFromNetwork refreshes shares correctly when the list of shares is not empty`() {
         every {
-            remoteShareDataSource.updateShare(
-                any(),
-                any(),
-                any(),
-                any(),
-                any(),
-                any()
+            remoteShareDataSource.getShares(
+                remoteFilePath = filePath,
+                reshares = true,
+                subfiles = false,
+                accountName = OC_ACCOUNT_NAME
             )
-        } returns shares[2]
+        } returns listOfShares
+
+        every {
+            localShareDataSource.replaceShares(listOfShares)
+        } returns listOf(1, 1, 1, 1)
```

Comment:
You're only replacing 2, right? That list should have just 2 long numbers

Reply:
That's right. I've removed two unnecessary long numbers 😄